frontend/packages/console-dynamic-plugin-sdk/src/coderefs/coderef-resolver.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-console */
+
+import * as _ from 'lodash';
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { SupportedExtension } from '../schema/console-extensions';
+import { RemoteEntryModule, EncodedCodeRef, CodeRef } from '../types';
+
+const isEncodedCodeRef = (obj: object): obj is EncodedCodeRef =>
+  _.isPlainObject(obj) &&
+  _.isEqual(Object.getOwnPropertyNames(obj), ['$codeRef']) &&
+  typeof (obj as EncodedCodeRef).$codeRef === 'string';
+
+export const filterCodeRefProperties = (properties: object) =>
+  _.pickBy(properties, isEncodedCodeRef) as { [propName: string]: EncodedCodeRef };
```

Comment:
Not sure it's fair to assume that all code refs appear as root properties.

Reply:
Maybe let's start with supporting code refs as root properties only and add nested code ref support in a follow-up?

Reply:
Will add a comment to implement support for code refs appearing on any level within the extension's properties object.